PyriteCore/src/display/texture.cpp

```diff
@@ -212,4 +212,116 @@ const SamplerState &TextureManager::getSampler(SamplerState::SamplerType type)
+	, m_arrayCount(count)
+	, m_isCubeArray(bIsCubeArray)
+{
+	if (bIsCubeArray) m_arrayCount *= 6;
+	D3D11_TEXTURE2D_DESC texDesc = {};
+	texDesc.Width = width;
+	texDesc.Height = height;
+	texDesc.MipLevels = 1;
+	texDesc.ArraySize = m_arrayCount;
+	texDesc.Format = bIsDepthOnly ? DXGI_FORMAT_R32_FLOAT : DXGI_FORMAT_R32G32B32A32_FLOAT;
+	texDesc.SampleDesc.Count = 1;
+	texDesc.Usage = D3D11_USAGE_DEFAULT;
+	texDesc.BindFlags = D3D11_BIND_SHADER_RESOURCE;
+	if (bIsCubeArray) texDesc.MiscFlags = D3D11_RESOURCE_MISC_TEXTURECUBE;
+
+	// Create the texture array resource
+	ID3D11Texture2D* texture;
+	DXTry(pyr::Engine::d3ddevice().CreateTexture2D(&texDesc, nullptr, &texture), "Could not create the texture associated with the array");
+	m_resource = texture;
+
+	D3D11_SHADER_RESOURCE_VIEW_DESC srvDesc = {};
+	srvDesc.Format = texDesc.Format;
+
+	if (bIsCubeArray)
+	{
+		srvDesc.ViewDimension = D3D11_SRV_DIMENSION_TEXTURECUBEARRAY;
+		srvDesc.TextureCubeArray.MostDetailedMip = 0;
+		srvDesc.TextureCubeArray.MipLevels = texDesc.MipLevels;
+		srvDesc.TextureCubeArray.First2DArrayFace = 0;
+		srvDesc.TextureCubeArray.NumCubes = count;
+	}
+	else
+	{
+		srvDesc.ViewDimension = D3D11_SRV_DIMENSION_TEXTURE2DARRAY;
+		srvDesc.Texture2DArray.MostDetailedMip = 0;
+		srvDesc.Texture2DArray.MipLevels = texDesc.MipLevels;
+		srvDesc.Texture2DArray.FirstArraySlice = 0;
+		srvDesc.Texture2DArray.ArraySize = texDesc.ArraySize;
+	}
+
+	DXTry(pyr::Engine::d3ddevice().CreateShaderResourceView(texture, &srvDesc, &m_textureArray), "Could not create a TextureArray.");
+}
+
+TextureArray::~TextureArray()
+{
+	DXRelease(m_textureArray);
+	DXRelease(m_resource);
+}
+
+void TextureArray::CopyToTextureArray(const std::vector<Texture>& textures, TextureArray& outArray)
+{
+	if (textures.empty()) return;
+	//if (!PYR_ENSURE(textures.size() == outArray.getArrayCount())) return;
+	if (!PYR_ENSURE(textures[0].getWidth() == outArray.getWidth())) return;
+	if (!PYR_ENSURE(textures[0].getHeight() == outArray.getHeight())) return;
+
+	ID3D11Texture2D* resource = static_cast<ID3D11Texture2D*>(outArray.getRawResource());
+
+	for (UINT i = 0; i < outArray.getArrayCount() && i < textures.size(); ++i)
+	{
+		// Copy the texture to the corresponding slice in the array
+		pyr::Engine::d3dcontext().CopySubresourceRegion(
+			resource,
+			D3D11CalcSubresource(0, i, 1),		// MipSlice = 0, ArraySlice = i
+			0, 0, 0,                            // Destination X, Y, Z
+			textures[i].getRawResource(), 
+			0,                                  // Source subresource index
+			nullptr);                           // No source box, copy entire resource
+	}
+}
+
+void TextureArray::CopyToTextureArray(const std::vector<Cubemap>& cubemaps, TextureArray& outArray)
+{
+	if (cubemaps.empty()) return;
+	// Ensure cubemaps and outArray dimensions match
+	//if (!PYR_ENSURE(cubemaps[0].getWidth() == outArray.getWidth())) return;
+	//if (!PYR_ENSURE(cubemaps[0].getHeight() == outArray.getHeight())) return;
```

Comment:
Is there a reason this is left-out?

Also both copy methods should assert `(!)m_isCubeArray`

Reply:
yeah, cause i never implemented the creation of the cubemap using a resolution, so this value isnt actually stored anyways, and would mean reading the TEXTURE2D_desc again